engine/v2/transaction/beef/source_transaction_resolver.go

```diff
@@ -0,0 +1,145 @@
+package beef
+
+import (
+	"fmt"
+
+	"github.com/bitcoin-sv/go-sdk/spv"
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/v2/transaction/errors"
+)
+
+// SourceTx represents a source transaction.
+type SourceTx struct {
+	Tx      *sdk.Transaction // Parsed transaction.
+	HadBeef bool             // Indicates if the transaction originated from a BEEF format.
+}
+
+// IsZero checks if the SourceTx is uninitialized.
+func (s SourceTx) IsZero() bool {
+	return s == SourceTx{}
+}
+
+// IsBeef checks if the SourceTx originated from a BEEF format.
+func (s SourceTx) IsBeef() bool {
+	return s.HadBeef
+}
+
+// SourceTxMap maps transaction IDs to SourceTx objects.
+type SourceTxMap map[string]SourceTx
+
+// Has checks if a transaction ID exists in the map.
+func (m SourceTxMap) Has(id string) bool {
+	_, ok := m[id]
+	return ok
+}
+
+// Add adds a TxQueryResult to the SourceTxMap after parsing it.
+func (m SourceTxMap) Add(q *TxQueryResult) error {
+	if q == nil {
+		return txerrors.ErrNilTxQueryResult
+	}
+
+	if q.IsBeef() {
+		tx, err := sdk.NewTransactionFromBEEFHex(*q.BeefHex)
+		if err != nil {
+			return fmt.Errorf("failed to parse BEEF transaction: %w", err)
+		}
+		m[q.SourceTXID] = SourceTx{Tx: tx, HadBeef: true}
+		return nil
+	}
+
+	if q.IsRawTx() {
+		tx, err := sdk.NewTransactionFromHex(*q.RawHex)
+		if err != nil {
+			return fmt.Errorf("failed to parse raw transaction: %w", err)
+		}
+		m[q.SourceTXID] = SourceTx{Tx: tx, HadBeef: false}
+		return nil
+	}
+
+	return txerrors.ErrTxQueryResultType
+}
+
+// Value retrieves the SourceTx for a given ID or returns an empty SourceTx.
+func (m SourceTxMap) Value(id string) SourceTx {
+	return m[id]
+}
+
+// TxQueryResultSlice represents a slice of TxQueryResult.
+type TxQueryResultSlice []*TxQueryResult
+
+// SourceTxMap converts the slice to a SourceTxMap.
+func (qq TxQueryResultSlice) SourceTxMap() (SourceTxMap, error) {
+	sourceTXs := make(SourceTxMap)
+	for _, q := range qq {
+		if err := sourceTXs.Add(q); err != nil {
+			return nil, fmt.Errorf("failed to add entry to source transaction map: %w", err)
```

Comment:
(Again) please use in this file `spverrors.Wrapf` instead, to keep consistency in code.